src/main/java/gyro/google/dns/PolicyResource.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    /**
+     * Controls whether logging is enabled for the networks bound to this policy. Defaults to no logging if not set. The value may be ``null``.
+     */
+    @Updatable
+    public Boolean getEnableLogging() {
+        return enableLogging;
+    }
+
+    public void setEnableLogging(Boolean enableLogging) {
+        this.enableLogging = enableLogging;
+    }
+
+    /**
+     * Unique identifier for the resource; defined by the server (output only). The value may be ``null``.
+     */
+    @Output
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * User assigned name for this policy. The value may be ``null``.
+     */
+    @Required
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * List of network names specifying networks to which this policy is applied. The value may be ``null``.
+     *
+     * @subresource gyro.google.dns.DnsPolicyNetwork
+     */
+    // XXX: https://github.com/perfectsense/gyro/issues/190
+    @Updatable
+    public List<DnsPolicyNetwork> getNetwork() {
+        return network;
+    }
+
+    public void setNetwork(List<DnsPolicyNetwork> network) {
+        this.network = network;
+    }
+
+    @Override
+    public boolean doRefresh() throws Exception {
+        Dns client = createClient(Dns.class);
+        Policy policy = client.policies().get(getProjectId(), getName()).execute();
+        return refreshFrom(policy);
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state) throws Exception {
+        Dns client = createClient(Dns.class);
+        Policy policy = new Policy();
+        DnsPolicyAlternativeNameServerConfig alternativeNameServerConfig = getAlternativeNameServerConfig();
+
+        if (alternativeNameServerConfig != null) {
+            policy.setAlternativeNameServerConfig(alternativeNameServerConfig.copyTo());
+        }
+        policy.setDescription(getDescription());
+        policy.setEnableInboundForwarding(getEnableInboundForwarding());
+        policy.setEnableLogging(getEnableLogging());
+        policy.setName(getName());
+        List<DnsPolicyNetwork> network = getNetwork();
+
+        if (network != null) {
+            policy.setNetworks(network
+                .stream()
+                .map(DnsPolicyNetwork::copyTo)
+                .collect(Collectors.toList()));
+        }
+        Policy response = client.policies().create(getProjectId(), policy).execute();
+        refreshFrom(response);
+    }
+
+    @Override
+    public void doUpdate(
+        GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        if (!(current instanceof PolicyResource)) {
+            throw new GyroException("Incompatible resource type! " + current.getClass().getName());
+        }
```

Comment:
Gyro would not be calling the `update()` with the wrong  `current` resource and the check is redundant

Reply:
Could you point me where types are checked?